src/com/github/igotyou/FactoryMod/Factorys/RepairFactory.java

```diff
@@ -242,23 +239,38 @@ public double getProductionTime() {
 
 	@Override
 	public ItemList<NamedItemStack> getRepairs() {
-		return rfp.getRepairMaterials();
+		if (mode == RepairFactoryMode.REPAIR) {
+			return rfp.getRepairMaterials();
+		} else {
+			return new ItemList<NamedItemStack>();
+		}
 	}
 
 	@Override
 	protected void recipeFinished() {
-		getInputs().removeOneFrom(getInventory());
+		log.finer("Recipe Finished");
+		//getInputs().removeOneFrom(getInventory()); // Don't double dip! Parent consume already ate recipe costs.
 		org.bukkit.inventory.ItemStack[] contents = getInventory().getContents();
 		for (int x = 0; x < contents.length; x++){
 			org.bukkit.inventory.ItemStack stack = contents[x];
-			if (stack == null)
+			if (stack == null) {
 				continue;
-			ItemStack s = CraftItemStack.asNMSCopy(stack);
-			s.setRepairCost(1);
-			getInventory().setItem(x, CraftItemStack.asBukkitCopy(s));
+			}
+			if (repairable(stack)) {
+				log.fine("Found repairable: " + stack.getType());
+				ItemStack s = CraftItemStack.asNMSCopy(stack);
+				s.setRepairCost(1);
```

Comment:
I don't want to make things cheaper if I can avoid it. If its not the best option we can just put online somthing that will piss people off. 

If its not relevant anyways go for it.